main.go

```diff
@@ -512,70 +514,22 @@ func downloadData(releases []string, filter string, storagePath string) {
-	topTestsWithBug := []*sippyprocessingv1.TestResult{}
-	all := result["all"]
-	withoutbugcount := 0
-	withbugcount := 0
-	// look at the top 100 failing tests, try to create a list of the top 20 failures with bugs and without bugs.
-	// limit to 100 so we don't hammer search.ci too hard if we can't find 20 failures with bugs in the first 100.
-	for i := 0; (withbugcount < 20 || withoutbugcount < 10) && i < 100 && i < len(all.TestResults); i++ {
-		test := all.TestResults[i]
-		test.BugList = bugCache.ListBugs(release, "", test.Name)
-
-		// we want the top ten test failures that don't have bugs associated.
-		// top test failures w/ bugs will be listed, but don't count towards the top ten.
-		if len(test.BugList) == 0 && withoutbugcount < 10 {
-			topTestsWithoutBug = append(topTestsWithoutBug, &test)
-			withoutbugcount++
-		} else if len(test.BugList) > 0 && withbugcount < 20 {
-			topTestsWithBug = append(topTestsWithBug, &test)
-			withbugcount++
-		}
-	}
-	return topTestsWithoutBug, topTestsWithBug
-}
-
-func (a *Analyzer) prepareTestReport(prev bool) {
-	byAll := util.SummarizeTestResults(a.RawData.ByAll, a.BugCache, a.Release, a.Options.MinTestRuns, a.Options.TestSuccessThreshold)
-	byPlatform := util.SummarizeTestResults(a.RawData.ByPlatform, a.BugCache, a.Release, a.Options.MinTestRuns, a.Options.TestSuccessThreshold)
-	byJob := util.SummarizeTestResults(a.RawData.ByJob, a.BugCache, a.Release, a.Options.MinTestRuns, a.Options.TestSuccessThreshold)
-	bySig := util.SummarizeTestResults(a.RawData.BySig, a.BugCache, a.Release, a.Options.MinTestRuns, a.Options.TestSuccessThreshold)
-
-	filteredFailureGroups := util.FilterFailureGroups(a.RawData.JobResults, a.BugCache, a.Release, a.Options.FailureClusterThreshold)
-	jobResults, infrequentJobResults := util.SummarizeJobRunResults(a.RawData.JobResults, byJob, a.BugCache, a.Release, a.Options.EndDay)
-
-	bugFailureCounts := util.GenerateSortedBugFailureCounts(a.RawData.JobResults, byAll, a.BugCache, a.Release)
-	bugzillaComponentResults := util.GenerateJobFailuresByBugzillaComponent(a.RawData.JobResults, byJob)
-
-	a.Report = sippyprocessingv1.TestReport{
-		Release:                        a.Release,
-		All:                            byAll,
-		ByPlatform:                     byPlatform,
-		ByJob:                          byJob,
-		BySig:                          bySig,
-		FailureGroups:                  filteredFailureGroups,
-		JobResults:                     jobResults,
-		InfrequentJobResults:           infrequentJobResults,
-		Timestamp:                      a.LastUpdateTime,
-		BugsByFailureCount:             bugFailureCounts,
-		JobFailuresByBugzillaComponent: bugzillaComponentResults,
-
-		AnalysisWarnings: a.analysisWarnings,
-	}
-
-	if !prev {
-		topFailingTestsWithoutBug, topFailingTestsWithBug := getTopFailingTests(byAll, a.Release, a.BugCache)
-		a.Report.TopFailingTestsWithBug = topFailingTestsWithBug
-		a.Report.TopFailingTestsWithoutBug = topFailingTestsWithoutBug
-	}
```

Comment:
assume that dropping this optimization is intentional (you are valuing the consistency of the report data over the optimization of not computing values we have no current use for) but your PR summary doesn't mention it, so was this intentional?

Reply:
> assume that dropping this optimization is intentional (you are valuing the consistency of the report data over the optimization of not computing values we have no current use for) but your PR summary doesn't mention it, so was this intentional?

Yes.  Knowing the previous list let's you flag things that have stayed in place.  Also, it no longer costs a BZ lookup.